### ISSUE TITLE
Caused by: java.lang.IllegalArgumentException: Unable to resolve the Configuration with the provided Issuer

### ISSUE THREAD

**Issue by rashidRizvi98** [ch4mpy/spring-addons#82]
I'm receiving following error when starting the app. any idea?

Caused by: java.lang.IllegalArgumentException: Unable to resolve the Configuration with the provided Issuer of "http://{my-vm's-ip}:{port}/realms/master"
        at org.springframework.security.oauth2.jwt.JwtDecoderProviderConfigurationUtils.getConfiguration(JwtDecoderProviderConfigurationUtils.java:166) ~[spring-security-oauth2-jose-6.0.0.jar:6.0.0]
        at org.springframework.security.oauth2.jwt.JwtDecoderProviderConfigurationUtils.getConfigurationForIssuerLocation(JwtDecoderProviderConfigurationUtils.java:79) ~[spring-security-oauth2-jose-6.0.0.jar:6.0.0]

**Comment by ch4mpy** [comment]
Yes, I have an idea why: `http://{my-vm's-ip}:{port}/realms/master/.well-known/openid-configuration` is not accessible from the host you start on. This can be because Keycloak is down or you have a network configuration issue. If you start the Spring resource-server from your IDE, try to open `openid-configuration` with your browser. Once you figured the right URI, report it to your Spring conf.

**Comment by rashidRizvi98** [comment]
Thanks for the reply. seems since keycloak v.17 /auth has been removed from context path(https://www.keycloak.org/migration/migrating-to-quarkus). My keycloak version was an old one. so mine was expecting /auth with issuer url. Now i have upgraded keycloak, seems that issue has been solved.

But now i am getting this one. I was following your "resource-server_with_oauthentication" tutorial

```
***************************
APPLICATION FAILED TO START
***************************

Description:

An attempt was made to call a method that does not exist. The attempt was made from the following location:

    com.c4_soft.springaddons.security.oauth2.config.synchronised.AddonsWebSecurityBeans.filterChain(AddonsWebSecurityBeans.java:174)

The following method did not exist:

    'java.lang.Object org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry.antMatchers(java.lang.String[])'

The calling method's class, com.c4_soft.springaddons.security.oauth2.config.synchronised.AddonsWebSecurityBeans, was loaded from the following location:

    jar:file:/home/xxxxx/.m2/repository/com/c4-soft/springaddons/spring-addons-webmvc-jwt-resource-server/6.0.0/spring-addons-webmvc-jwt-resource-server-6.0.0.jar!/com/c4_soft/springaddons/security/oauth2/config/synchronised/AddonsWebSecurityBeans.class

The called method's class, org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry, is available from the following locations:

    jar:file:/home/xxxxx/.m2/repository/org/springframework/security/spring-security-config/6.0.0/spring-security-config-6.0.0.jar!/org/springframework/security/config/annotation/web/configurers/ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry.class

The called method's class hierarchy was loaded from the following locations:

    org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer.ExpressionInterceptUrlRegistry: file:/home/xxxxx/.m2/repository/org/springframework/security/spring-security-config/6.0.0/spring-security-config-6.0.0.jar
    org.springframework.security.config.annotation.web.configurers.AbstractInterceptUrlConfigurer.AbstractInterceptUrlRegistry: file:/home/xxxxx/.m2/repository/org/springframework/security/spring-security-config/6.0.0/spring-security-config-6.0.0.jar
    org.springframework.security.config.annotation.web.configurers.AbstractConfigAttributeRequestMatcherRegistry: file:/home/xxxxx/.m2/repository/org/springframework/security/spring-security-config/6.0.0/spring-security-config-6.0.0.jar
    org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry: file:/home/xxxxx/.m2/repository/org/springframework/security/spring-security-config/6.0.0/spring-security-config-6.0.0.jar


Action:

Correct the classpath of your application so that it contains compatible versions of the classes com.c4_soft.springaddons.security.oauth2.config.synchronised.AddonsWebSecurityBeans and org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer$ExpressionInterceptUrlRegistry
```


Im using spring boot 3 and jdk17

This is my pom.xml


```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>3.0.0</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>
	<groupId>com.rashid</groupId>
	<artifactId>xxxxxxxi</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<name>xxxxxxx</name>
	<description>Demo project for Spring Boot</description>
	<properties>
		<java.version>17</java.version>
	</properties>
	<dependencies>
		<dependency>
			<groupId>org.springframework.security</groupId>
			<artifactId>spring-security-config</artifactId>
		</dependency>
		<dependency>
			<groupId>com.c4-soft.springaddons</groupId>
			<!-- use spring-addons-webflux-jwt-resource-server instead for reactive apps -->
			<artifactId>spring-addons-webmvc-jwt-resource-server</artifactId>
			<version>6.0.0</version>
		</dependency>
		<dependency>
			<groupId>com.c4-soft.springaddons</groupId>
			<!-- use spring-addons-webflux-test instead for reactive apps -->
			<artifactId>spring-addons-webmvc-jwt-test</artifactId>
			<version>6.0.0</version>
			<scope>test</scope>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-data-jpa</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>
		 <dependency>
                        <groupId>org.postgresql</groupId>
                        <artifactId>postgresql</artifactId>
                       <scope>runtime</scope>
                </dependency>
		<dependency>
			<groupId>org.projectlombok</groupId>
			<artifactId>lombok</artifactId>
			<optional>true</optional>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.springframework.security</groupId>
			<artifactId>spring-security-test</artifactId>
			<scope>test</scope>
		</dependency>		
	</dependencies>
	<build>
		<finalName>xxxxxxx</finalName>
		<plugins>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
				<configuration>
					<excludes>
						<exclude>
							<groupId>org.projectlombok</groupId>
							<artifactId>lombok</artifactId>
						</exclude>
					</excludes>
				</configuration>
			</plugin>
		</plugins>
	</build>
</project>
```

**Comment by ch4mpy** [comment]
With spring boot 3.0.0, use spring-addons 6.0.8.

6.0.0 was released at a time when only the first spring-boot 3 and spring-security 6 milestones were available. Spring has introduced breaking changes in between and you are facing one of them. The tutorials READMEs were a little behind, I just fixed that.

In case of doubt, refer to tutorials source-code. All tests have to pass before a release can be published.
